app/src/components/report/ReportEdit.tsx

```diff
@@ -0,0 +1,209 @@
+import { IReport } from '@/models';
+import { useCallback } from 'react';
+import Button from 'react-bootstrap/Button';
+import Form from 'react-bootstrap/Form';
+import Spinner from 'react-bootstrap/Spinner';
+import Stack from 'react-bootstrap/Stack';
+import ButtonLink from '../ButtonLink';
+
+export type ReportEditProps = {
+    report: Omit<IReport, 'reportId'> | undefined;
+    setReport: (newReport: IReport) => void;
+    submitLabel: string;
+    isSubmitLoading: boolean;
+    onSubmit: () => void;
+    cancelHref: string;
+};
+
+/** Report editor, used for create and edit flows */
+export default function ReportEdit({
```

Comment:
Nice job! Do you think we should set this up with Formik right away so it's easy to set up validations? Take a look at how Min incorporated Formik